Tests/FluentAssertions.Specs/Xml/XElementAssertionSpecs.cs

```diff
@@ -1228,5 +1228,185 @@ public void When_asserting_element_has_a_child_element_with_an_empty_name_it_sho
+            // Act
+            Action act = () => document.Should().HaveElement("child").Which.Should().BeSingle();
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage(
+                "Expected child to have a count of 1, but found 2.");
+        }
+
+        [Fact]
+        public void When_asserting_xElement_is_a_single_child_element_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                  </parent>");
+
+            // ReSharper disable once PossibleNullReferenceException
+            // Root is never null
+            var xElement = document.Root.Element("child");
+
+            // Act / Assert
+            xElement.Should().BeSingle();
+        }
+
+        [Fact]
+        public void When_asserting_xElement_is_not_a_single_child_element_it_should_fail()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // ReSharper disable once PossibleNullReferenceException
+            // Root is never null
+            var xElement = document.Root.Element("child");
+
+            // Act
+            Action act = () => xElement.Should().BeSingle();
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage(
+                "Expected child to have a count of 1, but found 2.");
+        }
+
+        [Fact]
+        public void When_asserting_a_null_xElement_to_be_single_it_should_fail()
+        {
+            // Arrange
+            XElement xElement = null;
+
+            // Act
+            Action act = () => xElement.Should().BeSingle();
+
+            // Assert
+            act.Should().Throw<InvalidOperationException>().WithMessage(
+                "Cannot assert the count if the element itself is <null>.");
+        }
+
+        #endregion
+
+        #region HaveCount
+
+        [Fact]
+        public void When_asserting_document_has_a_child_element_with_a_count_of_two_and_it_does_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // Act / Assert
+            document.Should().HaveElement("child").Which.Should().HaveCount(2);
+        }
+
+        [Fact]
+        public void When_asserting_document_has_a_child_element_with_a_count_of_two_and_it_does_not_it_should_fail()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                    <child />
+                  </parent>");
+
+            // Act
+            Action act = () => document.Should().HaveElement("child").Which.Should().HaveCount(2);
```

Comment:
I read this as: this document should have an element (one or more) named `child` which should have two child elements. But looking at the code I think it does something else. I suspect that what we you want is a `HaveElementCount` that takes a predicate or something like that.

Reply:
ok, I fixed it.